File: 1240.c
#include <stdio.h>
#include <string.h>

int main() {
    int n, i, tamP, tamS;
    char principal[1001], secundario[1001];

    scanf("%d", &n);
    for(i = 0; i < n; i++) {
        scanf("%s %s", principal, secundario);
        tamP = strlen(principal);
        tamS = strlen(secundario);

        if (tamS <= tamP && strcmp(principal + (tamP - tamS), secundario) == 0) {
            printf("encaixa\n");
        }else{
            printf("nao encaixa\n");
        }
    }

    return 0;
}

File: 1240.js
const fs = require("fs");
const input = fs.readFileSync(0, "utf8").trim().split(/\s+/);

let idx = 0;
const n = Number(input[idx++]);

let output = [];

for (let i = 0; i < n; i++) {
    const principal = input[idx++];
    const secundario = input[idx++];

    if (principal.endsWith(secundario))
        output.push("encaixa");
    else
        output.push("nao encaixa");
}

console.log(output.join("\n"));
